Added function for converting GCPs into a GeoTransform array

andyzimmerman1 opened the pull request:
I added this function because I needed a way to use the GCPs in an image to convert between pixel and geodetic coordinates. To enable this, this function converts the GCPs to a GDAL_GCP object and uses GDAL's GDALGCPsToGeoTransform to perform the actual conversion.

I did notice that the code for converting the []GCP -> C.goGCPList is the same as used in the SetGCPs function, and could perhaps be factored out into a helper function. I didn't do this yet, as I wasn't sure how to ensure that free always gets called for the `cIds` and `cInfos` CStringArrays.

tbonfort commented on godal.go:
```diff
@@ -4099,6 +4099,56 @@ func (ds *Dataset) SetGCPs(GCPList []GCP, opts ...SetGCPsOption) error {
+
+	gt := make([]C.double, 6)
+	cgt := (*C.double)(unsafe.Pointer(&gt[0]))
+	if C.godalGCPListToGeoTransform(gcpList, C.int(len(GCPList)), cgt) == 0 {
```
your error handling here is suboptimal. please use `godalWrap`/`cctx` in your c++ wrapper, and `createCgoWrapper` in your go code in order to use the standard error wrapping functionality provided by godal, as is done by all other wrapping code which is prone to returning an error

andyzimmerman1 replied:
Thanks for taking a look at this. I updated the error handling to use the `createCGOContext` to create the cctx which I then use for error handling. I also followed the convention that other functions use for creating a Opts struct that is passed in with the error handler. Could you please take a look and see if I used this paradigm correctly?

tbonfort replied:
@andyzimmerman1 lgtm thanks for the fixes. please return a [6]float64 instead of a *[6]float64 (unless there's a specific reason you chose to return a pointer, if so please explain)

andyzimmerman1 replied:
@tbonfort The only reason I was returning the pointer so that I could return `nil` for that return value in the case of an error. I though that was a standard convention, and noticed that pattern on other functions, such as `NewFeature`. I am happy to change it, though, as it is easy enough to instantiate the `[6]float64` for the error case

tbonfort reviewed (changes requested):
I see no urgent need to refactor the gcp list, the choice is yours. Please fix the error handling as detailed in the orther comment.

andyzimmerman1 commented:
Sorry for the delay at getting a test added. I updated the code to use the public GCP property names and also added tests that exercise passing in a custom error handler as an option.

tbonfort commented:
v0.0.10 tagged with this change

andyzimmerman1 commented:
@tbonfort - Thank you for the new tag!
